### PR TITLE
Add JOYSTICK_SERIAL_OUT system for outputting joystick data

### PR DESCRIPTION
**Pull request by GrumpyGopher**
For consideration:
Added new JOYSTICK_SERIAL_OUT system for outputting data via BlueRetro's serial port so that you can interface to other devices like the Arduino Pro Micro.

The data is outputted as a 30 character string beginning with the [ character and ending with the ] character.  This was done to allow for validation on the receiving side that the whole message was received.

Use Cases:
I'm using a BlueRetro HW1 Universal adapter.  This would allow me to use the same adapter for PC removing the need to resysnc controllers when switching back and forth between consoles and PC.  With this new feature I would be able to send the data to an Arduino Pro Micro which can emulate joysticks in Windows.

### CODE REVIEW

File: sdkconfig.defaults [hunk 1]
```diff
@@ -21,7 +21,7 @@ CONFIG_ESP_DEFAULT_CPU_FREQ_MHZ_240=y
 # Can't use SRAM1 for IRAM as it conflict with un-updatable bootloaders in the wild :@ .
 # CONFIG_ESP_SYSTEM_ESP32_SRAM1_REGION_AS_IRAM=y
 CONFIG_ESP_CONSOLE_UART_CUSTOM=y
-CONFIG_ESP_CONSOLE_UART_BAUDRATE=921600
+CONFIG_ESP_CONSOLE_UART_BAUDRATE=1000000
```
Comment (GrumpyGopher) [inline sdkconfig.defaults:24]: The default baud rate of 921600 does not play nice with Arduino Pro Micro.  The data comes across all garbled.  Tested with the following baud rates and had no problems connecting Arduino.
115200
250000
500000
1000000

File: main/adapter/adapter.h [hunk 2]
```diff
@@ -140,6 +141,24 @@ enum {
     PAD_RJ,
 };
 
+enum {
```
Comment (GrumpyGopher) [inline main/adapter/adapter.h:144]: this enum was added so I could minimize the number of characters needed to represent the 15 buttons of the controller.  With this new enum only 5 characters are needed.

**Comment by GrumpyGopher** [comment]
I also put together a couple Arduino sketches for the Arduino Pro Micro that go with this update.  The both utilize Arduino Joystick Library to emulate multiple joysticks with a single Arduino Pro Micro.  I made a 6 player and 4 player version.  I know BlueRetro can handle 7 controllers but I don't know if Arduino could handle that and I couldn't think of any 7 player games. The 6 player version requires disabling CDC which some people may not want to do but the 4 player version will work even if CDC is enabled.

After uploading the 6 player version and setting up the controllers in Steam I was able to run TMNT: Shredder's Revenge with all 6 controllers.

Github for the sketches:
https://github.com/GrumpyGopher/BlueRetroToArduinoJoystick

**Comment by darthcloud** [comment]
Thanks for sharing!

I'm planning to define a SPI interface for interfacing with other MCU or FPGA sometimes this year.
That will be the official way to integrate with BlueRetro.

I'm very sorry but for that reason I will not merge this PR. I don't want to start documenting and supporting this if I'm going to replace it anyway soon.

But I'm glad that you dig into BlueRetro code and made something useful.

**Comment by GrumpyGopher** [comment]
It was fun to put together but I look forward to seeing the SPI implementation.  In the meantime is there a way I can either run the web config locally so I can add the system option for myself or maybe a way to add my system to the auto config?  It would just make it easier for me to toggle between console and PC while waiting for SPI to be released.

Thanks

**Comment by GrumpyGopher** [comment]
Thank you so much, forking the repo worked perfectly

**Pull request closed without merging**